### DISCUSSION TITLE
A few calibration questions ( dim leds and cec )

### DISCUSSION THREAD

**Discussion by jingato** [awawa-dev/HyperHDR#744] [Q&A]
Hey all,

So I finally got my system setup and everything is running great, minus a couple small issues. I am trying to work them out now though. I haven't found any specifics in the wiki or discussions though, so figured I would make a post.

For reference, My system consists of the following
Raspberry Pi 4
pimoroni plasma 2040 Driver
SK6812 LED Strip
Elgato HD60X Capture Device
HyperHDR 20 beta

So the first issue I am having is that whenever the HDR toggle is turned on I get really dim LEDs. When I toggle it off, they look bright. I have all the proper setup done on the "video capturing" section. I'm using NV12, Quarter Frame Mode, HDR to SDR Tone mapping. For the LED hardware I made sure to enable the AWA protocol and the Rp2040 handshake. I also setup my LUT by downloading the preconfigured LUT table for my capture device. In the LUT table section when I apply it to the image, the colors do look good in the preview. This seems to be the case for both SDR content AND HDR content. I would assume I would want it off for SDR content and on for HDR content though, right?

Which brings me to my next question. Is it expected for the HDR toggle to happen automatically when switching between SDR / HDR? I see that there is an option to enable cec in the video capture section and also offers an HDR / SDR key to set. However, I turned it on and I am not seeing any logging for receiving cec key commands. However, I am seeing one ominous log saying "FIXME: LG seems to have bugged out. resetting to 'in transition standby to on'. the return button will not work". This looks to be an issue with libcec as I see others online reporting this who are not using HyperHDR. I don't know if it is a red herring though as I don't need to use any power or standby features, only the HDR / SDR signaling. For reference, I plugged an HDR cable from my TV's (ARC) HDMI input directly to my Raspberry Pi. I was going to start looking into the source a bit more and debugging it, but I figured I would make sure I am not wasting my time first and that this is even possible. 

Full logs related to CEC
2024-01-22T22:28:00.216Z [CEC] Opening libCEC library.
2024-01-22T22:28:00.216Z [CEC] CEC object created
2024-01-22T22:28:00.216Z [CEC] Starting CEC handler.
2024-01-22T22:28:03.623Z [CEC] Success. CEC adapter type 'Linux' is ready (/dev/cec0, /sys/devices/platform/soc/fef00700.hdmi/cec0)
2024-01-22T22:28:03.623Z [CEC] Success: libCEC library loaded.
2024-01-22T22:28:03.317Z [CEC] CEC library warning: FIXME: LG seems to have bugged out. resetting to 'in transition standby to on'. the return button will not work

I did find [this ](https://github.com/Pulse-Eight/libcec/issues/567)regarding that error, but the suggested fix looks to be Kodi specific if I am correct. I don't see any file like that for HyperHDR(??)

Thanks!

John

**Comment by awawa-dev** [comment]
Hi
> the LUT table section when I apply it to the image, the colors do look good in the preview. 

You should check the live preview in HyperHDR (I suspect that you are using a tool that generates LUT or import it, not using the downloaded one). Could you provide screenshot from HyperHDR with some bright (on the TV) HDR scene with tone mapping enabled and disable? Can you check video format colorspace you are using in the video player (ex. HDR10/DolbyVision and RGB444,YUV444, YUV420: full range or limited)? You probably also have information on your TV in one of the menus about the video format it will receive: this information would also be useful.

> I see that there is an option to enable cec in the video capture section and also offers an HDR / SDR key to set. 

Unfortunately libCEC is one leg in the grave, especially when Bookworm OS is used: https://forums.raspberrypi.com/viewtopic.php?t=348779
Nevertheless power off/on events should work at least on Bullseye OS, but as @incarvr6 also reported CEC keys may not work.

**Reply by incarvr6** [reply]
Yes I managed to make CEC keys work on Bullseye. Had to modify config file and install cec-client package.
Functionality is basic, all it gives is option to map some spare keys on TV remote to manually switch on or off HDR mapping. But even that only works when Pi is selected as source on TV.
Only advantage of CEC right now is, it is recognised by HyperHDR when TV is switched on or off HyperHDR led output is switched as well.

**Comment by jingato** [comment]
Ok, this is interesting. I went to use the live preview like you mentioned and I took a screenshot of it with HDR turned off. The turned HDR on and went to take a screenshot and it actually just says "no signal". This is when displaying HDR or SDR content. I also noticed that in the remote control section the active source under source selection disappears when HDR is switched on.

I'm not sure how to get that information from my TV, it is an LG oled65b6p, but if I look in my Xbox menu it says my TV supports HDR10, but not dolby vision and I have "allow ycc 4:2:2" set to on. When I click the info button on my TV it just says 2160p HDR bt.2020.

However, if I press the info button on my Denon receiver, which the HDMI is going through, it gives me some more info. It says
Resolution => 4K60
HDR => HDR10
Color Space => YCbCr 4:2:2/BT.2020
Pixel Depth => 12 bits

When I turn on the HDR switch ( doing it under the remote control section ) then this is the output I see in the logs regarding the capture
2024-01-23T23:42:52.463Z [V4L2:ELGATO HD60 X:] Video cache: size: 2, hits: 3600, needed: 0, cleanup: 1, limit: 8
2024-01-23T23:43:52.462Z [PERFORMANCE] [USB: FPS = 60.00, decoding = 0ms, frames = 3600, invalid = 0, mode = direct]

I also tried changing the video grabbers format from nv12 to yuyv but that didn't seem to help at all.

That's unfortunate about the cec. I read that it also has specific isssues with LG tv's too so I'm double screwed there. It's no biggie.

**Comment by awawa-dev** [comment]
> The turned HDR on and went to take a screenshot and it actually just says "no signal".

This is a hardware problem for sure. I thought you had brightness problems when you clicked HDR tone mapping in HyperHDR and not with switching between SDR/HDR on the video source/player. I don't know why, maybe HDCP is also activated with HDR and that disables Elgato capturing (only no signal screen is returned).

> When I click the info button on my TV it just says 2160p HDR bt.2020.

The information you provided from LG TV is sufficient and I also use it (just as a curiosity, LG has a more accurate signal report hidden [link](https://www.avsforum.com/threads/official-2019-lg-c9-consolidated-info-faq-troubleshooting-firmware-more.3075344/post-58698658) )

**Comment by jingato** [comment]
Hmm, I tried a few different sources and they all do the same thing. I don't believe it is an HDCP thing because the HDCP wasn't causing a black image, it was just causing it to continually reset and blink in and out. Plus, I got my splitter today and hooked that up and now HDCP content displays and captures just fine. It's only when I turn the HDR switch on. This is whether the TV is outputting HDR or not.

So what does the HDR switch in the web configuration do exactly? Is it just changing the software side or is it triggering something on the capture device? I guess it is possible I have a bad HD60X. What would be the best way to test that? I could connect it to my PC usb instead and try to capture through their capture utility software, but is there anything special I would need to do to verify that or just capture with an HDR source?

**Reply by awawa-dev** [reply]
That happens when you enable HDR in HyperHDR or play HDR content? If you have some video visible in the live preview and then you enable tone mapping in HyperHDR and it becomes completely black, then it may indicate broken LUT table. Did you try to perform calibration and interrupted the process? More should be in logs including the path to that LUT. If it is in your home user directory, it's a custom LUT.

**Reply by jingato** [reply]
Ah, ok. Yeah, it happens when I toggle the the enable HDR in HyperHDR. It can be SDR or HDR source content, it doesn't matter. I just clicked the download button for the LUT under the Elgato DH60X. Now when I go to the "LUT->Download section it says "Custom LUT path is: /root/.hyperhdr/lut_lin_tables.3d. Custom LUT is found" I can try deleting the LUT file and trying it, then downloading it again, and if still nothing, I can do the custom calibration. I didn't attempt the custom one since I figured the download for my capture device was available. I'll report back.

**Reply by awawa-dev** [reply]
Ok, you after re-dowload LUT make sure you check HyperHDR full logs to verify if HyperHDR managed to load it. Otherwise app may create LUT table by its own as a rescue, but tone mapping won't work.

**Comment by jingato** [comment]
Awesome, deleting and redownloading did the trick....and man, the colors look phenomenal! Not sure what went wrong with the previous download, perhaps when I changed from running as pi to running as root perhaps.

Thanks again!
